### ISSUE TITLE
Out of date Meghanada server install URL

### ISSUE THREAD

**Issue by dfpetrin** [hlissner/doom-emacs#4979]
**What did you expect to happen?**
Meghanada server would install after opening first java file.

**What actually happened?**
*meghanada-install-error* buffer popped up with:

Error: Not found https://dl.bintray.com/mopemope/meghanada/meghanada-setup-0.0.2.jar

**Additional details:**

If you go to this URL manually it returns "Forbidden!"

The manual install instructions on the Meghanada Github repo (https://github.com/mopemope/meghanada-emacs/) use a different URL: https://github.com/mopemope/meghanada-server/releases/download/v1.0.13/meghanada-setup-0.0.2.jar

Based on this past issue for meghanada (https://github.com/mopemope/meghanada-emacs/issues/84), it sounds like the version of meghanada doom is pinned to is pointing to an outdated server install URL.

For now I'm just following the manual install instructions.

**Steps to reproduce:**
1. Install emacs-mac
2. Install doom
3. Enable java and +meghanada in init
4. Open a java file

**System information:**
https://pastebin.com/sFE9MXsh

**Comment by dfpetrin** [comment]
Oh, no backtrace available since the meghanada server install program catches the error itself and writes to its own error buffer.

**Comment by hlissner** [comment]
This appears to be [an upstream issue](https://github.com/mopemope/meghanada-emacs/blob/1e41f7f2c7a172e9699f3557c97c3f39a149bfc2/meghanada.el#L439). Meghanada hasn't updated it's download urls and offers no variable for us to customize.

I've created [a PR with a fix](https://github.com/mopemope/meghanada-emacs/pull/177). I'll leave this open until it's been merged and I've bumped the meghanada package.

In the meantime, this will work around the issue:

```elisp
(defadvice! +java-fix-meghanada-download-url-a (args)
  :filter-args #'meghanada--download-from-url
  (cons (if (equal (format
                    "https://dl.bintray.com/mopemope/meghanada/meghanada-setup-%s.jar"
                    meghanada-setup-version)
                   url)
            (format
             "https://github.com/mopemope/meghanada-server/releases/download/v%s/meghanada-setup-%s.jar"
             meghanada-version
             meghanada-setup-version)
          url)
        args))
```

**Comment by hlissner** [comment]
The fix was merged upstream and meghanada was bumped in 464ce5e, which resolves this issue. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!

**Comment by dfpetrin** [comment]
Wow, such a rapid resolution! Thanks! I removed the downloaded .jar files and relaunched Doom and opened a .java file, installed as expected. Confirmed resolved.

Of course I've now found meghanada is the latest addition to my list of "Reasons never to early adopt", since it currently doesn't seem to support the Apple M1 chips, and all kinds of madness is happening when I try with a Rosetta/Intel install of java. *sigh*

Or this is just punishment for trying to do java in something other than Eclipse/IntelliJ 😁
